packages/spl-token/src/domain/main.ts

```diff
@@ -0,0 +1,147 @@
+import { Token, solanaPrivateRPCRoundRobin } from '@aleph-indexer/core'
+import {
+  SPLAccountBalance,
+  SPLAccountHoldings,
+  SPLTokenEvent,
+  SPLTokenInfo,
+  SPLTokenType,
+} from '../types.js'
+import { discoveryFn } from '../utils/discovery.js'
+import { AccountHoldingsFilters, MintEventsFilters } from './types.js'
+import { TOKEN_PROGRAM_ID } from '../constants.js'
+
+export default class MainDomain
+  extends IndexerMainDomain
+  implements IndexerMainDomainWithDiscovery, IndexerMainDomainWithStats
+{
+  protected tokens: Record<string, SPLTokenInfo> = {}
+
+  constructor(protected context: IndexerMainDomainContext) {
+    super(context, {})
+  }
+
+  async updateStats(now: number): Promise<void> {
+    console.log('Method not implemented.')
+  }
+
+  async discoverAccounts(): Promise<AccountIndexerRequestArgs[]> {
+    const init = {
+      account: '',
+      index: {
+        transactions: {
+          chunkDelay: 0,
+          chunkTimeframe: 1000 * 60 * 60 * 24,
+        },
+        content: false,
+      },
+    }
+    return [init]
+  }
+
+  async init(...args: unknown[]): Promise<void> {
+    super.init(...args)
+    const { accounts, mints } = await discoveryFn()
+
+    await Promise.all(
+      accounts.map(async (account: string) => {
+        const connection = solanaPrivateRPCRoundRobin.getClient()
+        const mint = await Token.getTokenMintByAccount(
+          account,
+          connection.getConnection(),
+        )
+        await this.addToken(mint)
+
+        const options = {
+          account,
+          meta: { type: SPLTokenType.Account, mint: mint },
+          index: {
+            transactions: {
+              chunkDelay: 0,
+              chunkTimeframe: 1000 * 60 * 60 * 24,
+            },
+            content: false,
+          },
+        }
+        await this.context.apiClient.indexAccount(options)
+        this.accounts.add(account)
+      }),
+    )
+    await Promise.all(
+      mints.map(async (mint: string) => {
+        await this.addToken(mint)
+        const options = {
+          account: mint,
+          meta: { type: SPLTokenType.Mint, mint },
+          index: {
+            transactions: {
+              chunkDelay: 0,
+              chunkTimeframe: 1000 * 60 * 60 * 24,
+            },
+            content: false,
+          },
+        }
+        await this.context.apiClient.indexAccount(options)
+        this.accounts.add(mint)
+      }),
+    )
+  }
+
+  async getTokens(): Promise<Record<string, SPLTokenInfo>> {
+    return this.tokens
+  }
+
+  async getTokenHolders(account: string): Promise<SPLAccountBalance[]> {
+    return (await this.context.apiClient.invokeDomainMethod({
+      account,
+      method: 'getTokenHolders',
+    })) as SPLAccountBalance[]
+  }
+
+  async getMintEvents(
+    account: string,
+    filters: MintEventsFilters,
+  ): Promise<SPLTokenEvent[]> {
+    return (await this.context.apiClient.invokeDomainMethod({
+      account,
```

Comment:
this is load balancing and partitioning by mint account, but in other methods the balancing is being made by token account. Review this, because with the current configuration there are two big issues:
1. Querys by mint without token account will not retrieve all the mint events
2. Some querys by mint and account will not retrieve anything in case that the mint and the account addresses generate different partition key hashes